extension/diffheader.js

```diff
@@ -21,6 +21,46 @@ window.diffFileHeader = (() => {
 		};
 	})();
 
+	const maxPxAvailable = () => {
+		// Unfortunately can't cache this value, as it'll change with the browsers zoom level
+		const filenameLeftOffset = $('.diff-toolbar-filename').get(0).getBoundingClientRect().left;
+		const diffStatLeftOffset = $('.diff-toolbar-filename + .right').get(0).getBoundingClientRect().left;
+
+		return diffStatLeftOffset - filenameLeftOffset;
+	};
+
+	const parseFileDetails = filename => {
+		const folderCount = (filename.match(/\//g) || []).length;
+		const [, basename] = (filename.match(/(?:\/)([\w\.-]+)$/) || []);
+		const [, topDir] = (filename.match(/^([\w\.-]+)\//) || []);
```

Comment:
I'll note that this assumes the root in a diff never starts with a `/` - I'm *pretty* sure that's accurate, but correct me if I'm wrong and I'll update.